chartlets.js/src/lib/components/Plot.tsx

```diff
@@ -2,32 +2,109 @@ import { VegaLite } from "react-vega";
 
 import { type PlotState } from "@/lib/types/state/component";
 import { type ComponentChangeHandler } from "@/lib/types/state/event";
+import {
+  isString,
+  isTopLevelSelectionParameter,
+  type SignalHandler,
+} from "@/lib/types/state/vega";
+import { useCallback, useMemo } from "react";
 
 export interface PlotProps extends Omit<PlotState, "type"> {
   onChange: ComponentChangeHandler;
 }
 
 export function Plot({ id, style, chart, onChange }: PlotProps) {
+  /*
+  Here, we loop through all the params to create map of signals which will be then used to create the map of
+  signal-listeners
+  */
```

Comment:
Why? Please justify, why we need to that. What is the goal?

Reply:
Because not all params are event-listeners, and we need to identify which of the params are event-listeners. 
This loop just does that.

Later in the code, we then see which handlers do we have so that we can create those listeners with the names specified in the event-listener object.

I will add this info as well to the comment.